lib/jellyfish/component/hls/manager.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Jellyfish.Component.HLS.Manager do
+  @moduledoc """
+  Module responsible for HLS processing.
+  """
+
+  use GenServer
+
+  require Logger
+
+  alias Jellyfish.Room
+
+  @hls_extensions [".m4s", ".m3u8", ".mp4"]
+  @playlist_content_type "application/vnd.apple.mpegurl"
+
+  @spec start(Room.id(), pid(), String.t(), map()) :: :ok
+  def start(room_id, engine_pid, hls_dir, hls_options) do
+    {:ok, _pid} = GenServer.start(__MODULE__, [room_id, engine_pid, hls_dir, hls_options])
+    :ok
+  end
+
+  @impl true
+  def init([room_id, engine_pid, hls_dir, hls_options]) do
+    Process.monitor(engine_pid)
+    Logger.info("Initialize s3 uploader")
+
+    {:ok, %{room_id: room_id, engine_pid: engine_pid, hls_dir: hls_dir, hls_options: hls_options}}
+  end
+
+  @impl true
+  def handle_info(
+        {:DOWN, _ref, :process, engine_pid, _reason},
+        %{engine_pid: engine_pid} = state
+      ) do
+    unless is_nil(state.hls_options.s3) do
+      state.hls_options.s3
+      |> create_aws_config()
+      |> upload_to_s3(state.hls_dir, state.hls_options.s3.bucket, state.room_id)
+    end
```

Comment:
Keep in mind this is blocking so any other callback won't be called until we finish uploading